src/Apps/Order/Components/ExpressCheckout/__tests__/ExpressCheckoutUI.jest.tsx

```diff
@@ -205,6 +210,87 @@ describe("ExpressCheckoutUI", () => {
     expect(env.mock.getAllOperations()).toHaveLength(0)
   })
 
+  it("omits shipping data when order is pickup", async () => {
+    const { mockResolveLastOperation, env } = renderWithRelay({
+      Order: () => ({
+        ...orderData,
+        fulfillmentOptions: [
+          {
+            type: "PICKUP",
+            amount: {
+              minor: 4200,
+              currencyCode: "USD",
+            },
+            selected: true,
+          },
+        ],
+      }),
+    })
+
+    fireEvent.click(screen.getByTestId("express-checkout-confirm"))
+
+    const { operationName, operationVariables } =
+      await mockResolveLastOperation({
+        updateOrderPayload: () => ({
+          orderOrError: {
+            __typename: "OrderMutationSuccess",
+            order: orderData,
+          },
+        }),
+      })
+    expect(operationName).toBe("useUpdateOrderMutation")
+    expect(operationVariables.input).toEqual({
+      id: "a5aaa8b0-93ff-4f2a-8bb3-9589f378d229",
+      buyerPhoneNumber: "1234567890",
+      buyerPhoneNumberCountryCode: null,
+      creditCardWalletType: "APPLE_PAY",
+      paymentMethod: "CREDIT_CARD",
+      shippingAddressLine1: "401 Broadway",
+      shippingAddressLine2: null,
+      shippingCity: "New York",
+      shippingCountry: "US",
+      shippingName: "Buyer Name",
+      shippingPostalCode: "10013",
+      shippingRegion: "NY",
```

Comment:
I was confused by this mutation that sends shipping address, and I think this is actually a 🐛 . We should avoid updating order with shipping information (even tho it's provided by Apple Pay UI) when it's pickup (which was previously [ensured on the backend](https://github.com/artsy/exchange/blob/f927146e4a65dbfd1d1c3d59f37013f20506dd80/lib/order_shipping.rb#L6-L23)). The order ends up in a wrong state with fulfillment type `pick up` _and_ shipping address.

![Screenshot 2025-04-23 at 5 10 12 PM](https://github.com/user-attachments/assets/95d88822-4292-4f16-a490-d2570e24cc43)